### PR TITLE
Bots no longer require PAIs to become sapient

### PR DESCRIPTION
**Pull request by Jacquerel**
## About The Pull Request

We were talking in the coder channel about what the role of a pAI is, with a general conclusion that as the name would suggest they should be _personal assistants_.
This means they should be sticking around their owner, not wandering away as a holochassis or in the body of a bot.
The former is a matter for a future PR, the latter I am addressing here.

What we also discussed is that clearly some people _want_ to respawn as a weird quasi-useless mob which wanders aimlessly around the station. That seems like a fine thing to exist, but it shouldn't be a pAI.

Resultingly: pAI cards can no longer be placed inside bots.
However, you also no longer need to place pAI cards inside bots in order for them to become sapient, it's a simple toggle on the bot control menu. Enabling this option will poll ghosts 
Toggling the "personality matrix" off while a bot is being controlled by a ghost will ghost them again, so if they're annoying they're not that hard to get rid of.

image

Mobs which couldn't have a pAI inserted don't have this option. Specifically securitrons, ED-209, and Hygienebots (for some reason).

Perhaps most controversially, any bots which are present on the station when the map loads will have this setting enabled by default. We will see if players abuse this too much and need their toys taken away, I am hoping they can be trusted.

Additionally, as part of this change, mobs you can possess now appear in the spawners menu.
image
Here is an unusually populated example.

Oh also in the process of doing this I turned the regal rat "click this to become it" behaviour into a component because it seems generally useful.

## Why It's Good For The Game

Minor stuff for dead players to do if they want to interact with living players instead of observe.
Shift pAI back into a more intended role as a personal assistant who hangs around with their owner, rather than just a generic respawn role.

## Changelog

:cl:
add: PAIs can no longer be inserted into Bots
add: Bots can now have their sapience toggled by anyone with access to their settings panel
add: Bots which exist on the map at the start of the round automatically have this setting enabled
qol: Bots, Regal Rats, and Cargorilla now appear in the Spawners menu if you are dead
qol: Bots can be renamed from their maintenance panel
/:cl:

### CODE REVIEW

**Comment by DaydreamIQ** [comment]
Are beepskies gonna be the sole exception here? I think it'd be a terrible idea if sec could somehow mass produce sentient ones and absolutely style on any antag who doesn't pack an emp.

**Comment by IndieanaJones** [comment]
So here's a question for you: if I emag a sentient bot, are they an antagonist now?

**Comment by Jacquerel** [comment]
> Are beepskies gonna be the sole exception here? I think it'd be a terrible idea if sec could somehow mass produce sentient ones and absolutely style on any antag who doesn't pack an emp.

Turns out its just disabled on all security bots and not just ones who are called beepsky

**Comment by SomeoneYouProbablyKnow** [comment]
Mulebots are included? Unbelievable amounts of faith in the playerbase.

Would be nice if EMAG did something to these fellas. Lock down the personality matrix and make em follow your orders, just like borgs. But maybe that's too powerfull and it should just kick em out to prevent antag rolling as roombas

**Comment by TheVekter** [comment]
Hygeinebots can't have pAIs inserted because their emagged version spews fire at whoever they try to clean.

Mulebots should probably not be included here, that sounds like a nightmare waiting to happen.

**Comment by SomeoneYouProbablyKnow** [comment]
> Hygeinebots can't have pAIs inserted because their emagged version spews fire at whoever they try to clean.
> 
> Mulebots should probably not be included here, that sounds like a nightmare waiting to happen.

Tbf you gotta hack the mulebot first. And them being sentient from time to time might encourage cargo to do more deliveries.

Cleanbots on the other hand are scary, you don't need to hack it, just slap a knife on top of it.

**Comment by Jacquerel** [comment]
> Mulebots are included? Unbelievable amounts of faith in the playerbase.

MULEbots under player control can't trample people, which was an existing functionality for pAI controlled ones.

> Would be nice if EMAG did something to these fellas. Lock down the personality matrix and make em follow your orders, just like borgs. But maybe that's too powerfull and it should just kick em out to prevent antag rolling as roombas

Yeah I probably do need to add something additional for emag handling... I am, inclined to err on the side of "it disables the personality" because cyborgs can be bad enough about emags that bots without lawsets are going to be worse.

**Comment by SomeoneYouProbablyKnow** [comment]
> MULEbots under player control can't trample people, which was an existing functionality for pAI controlled ones.

Huh, neat. Sad, but neat.

> Yeah I probably do need to add something additional for emag handling... I am, inclined to err on the side of "it disables the personality" because cyborgs can be bad enough about emags that bots without lawsets are going to be worse.

Fair, in that case you can add hygiene bots to the list as their emagged state is the problem not their normal one.

**Comment by BlueMemesauce** [comment]
> Shift pAI back into a more intended role as a personal assistant who hangs around with their owner, rather than just a generic respawn role.

I guess you're technically correct that the specific pAI role can no longer do these things, but in reality all you're doing is calling it a different name and removing the pAI step. The actual function still exists, except worse because bots don't have to have to follow anyone's orders and can't act antagonistically anymore. No more commanding a hacked mulebot or knifed cleanbot to help you kill people. So it's actually going to do the opposite of what you said and make the generic respawn role thing even more prevalent since you entirely removed the ability of them to follow orders, all they can do is non-antagonistically perform their function. Yes it technically won't fall under pAI, but is that a good thing? 


> Cleanbots on the other hand are scary, you don't need to hack it, just slap a knife on top of it.

> Fair, in that case you can add hygiene bots to the list as their emagged state is the problem not their normal one.

You're confusing cleanbots and hygienebots, they're different things.

**Comment by SomeoneYouProbablyKnow** [comment]
Hygine bots are flying showers / fire showers when emmaged

Clean bots are buckets on wheels that have terrible pathfinding. You can slap a knife on those, unless i missed that getting changed as well

**Comment by BlueMemesauce** [comment]
> Hygine bots are flying showers / fire showers when emmaged
> 
> Clean bots are buckets on wheels that have terrible pathfinding. You can slap a knife on those, unless i missed that getting changed as well

yea i was the one that got confused

**Comment by Jacquerel** [comment]
> I guess you're technically correct that the specific pAI role can no longer do these things, but in reality all you're doing is calling it a different name and removing the pAI step. The actual function still exists, except worse because bots don't have to have to follow anyone's orders and can't act antagonistically anymore. No more commanding a hacked mulebot or knifed cleanbot to help you kill people. So it's actually going to do the opposite of what you said and make the generic respawn role thing even more prevalent since you entirely removed the ability of them to follow orders, all they can do is non-antagonistically perform their function. Yes it technically won't fall under pAI, but is that a good thing?

To be clear the idea wasn't "eliminate players who want to respawn as a creature and fuck around".
It was "make them not do it while playing as a pAI".

**Comment by TheVekter** [comment]
> You can slap a knife on those, unless i missed that getting changed as well

I believe this is still possible. Please don't remove it; it doesn't do that much damage and is extremely funny.

**Comment by SomeoneYouProbablyKnow** [comment]
> I believe this is still possible. Please don't remove it; it doesn't do that much damage and is extremely funny.

It is really funny. Man even steals tittles from bots he kills last time i checked. It's actually pretty strong against people (untill they figure out how to throw chairs) as you don't have to click, people just trip on you, but... it's REALLY funny.

> tldr : Tators no longer get to get bot accomplices via pais.

I mean, i sure that's something that can be done about that. What about this : 
Emag kicks out the ghost
The tator can re enable the matrix
A ghost can come in loyal to the emager

Everyone got options, everyone is happy

**Comment by DATA-xPUNGED** [comment]
I feel like pAIs should still be able to be placed inside bots, for people who do want a personal assistant that also happens to be a bot

**Comment by BlueMemesauce** [comment]
> To be clear the idea wasn't "eliminate players who want to respawn as a creature and fuck around". It was "make them not do it while playing as a pAI".

I just don't really understand the point of caring about the name that much? I feel like the actual gameplay itself should be what you're focusing on. I don't really know how to explain this any better, but it's like you're basically just creating a new tool called dwAI (Do-whatever AI) that can be used instead of pAI. In the end, all this change really accomplishes is removing the feature of being a murder bot like I said before. It may solve the problem of the role not being named correctly, but it doesn't solve the actual problem of people taking over bots and fucking off as them.

**Comment by Jacquerel** [comment]
> I just don't really understand the point of caring about the name that much? I feel like the actual gameplay itself should be what you're focusing on. I don't really know how to explain this any better, but it's like you're basically just creating a new tool called dwAI (Do-whatever AI) that can be used instead of pAI. In the end, all this change really accomplishes is removing the feature of being a murder bot like I said before. It may solve the problem of the role not being named correctly, but it doesn't solve the actual problem of people taking over bots and fucking off as them.

We don't care about people taking over bots and fucking off as them, we care about people taking pAI cards and fucking off as them. The problem you're identifying isn't the problem I am setting out to solve, nor frankly do I think it is a big deal.
If people don't want their cleanbot to wander off they'll just not turn on this feature.

**Comment by Jacquerel** [comment]
As a side note it turns out that nobody ever bothered giving hygienebots the ability to do anything at all when player controlled, so if someone wanted to change that they can do it in a future PR.

File: code/datums/components/_component.dm [hunk 1]
```diff
@@ -52,10 +52,14 @@
...
 	var/list/arguments = raw_args.Copy(2)
-	if(Initialize(arglist(arguments)) == COMPONENT_INCOMPATIBLE)
+	var/initialised = Initialize(arglist(arguments))
+	if(initialised == COMPONENT_INCOMPATIBLE)
 		stack_trace("Incompatible [type] assigned to a [parent.type]! args: [json_encode(arguments)]")
 		qdel(src, TRUE, TRUE)
 		return
+	if(initialised == INITIALIZE_HINT_QDEL)
+		qdel(src, TRUE, TRUE)
+		return
```
Comment (Mothblocks) [inline code/datums/components/_component.dm:62]: switch if

File: code/datums/components/ghost_direct_control.dm [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+	datum/callback/after_assumed_control,
+)
+	. = ..()
+	if (!isliving(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	if (!(GLOB.ghost_role_flags & GHOSTROLE_SPAWNER))
+		return INITIALIZE_HINT_QDEL
+
+	src.poll_role_string = isnull(poll_role_string) ? "[parent]" : poll_role_string
```
Comment (Mothblocks) [inline code/datums/components/ghost_direct_control.dm:33]:
```suggestion
	src.poll_role_string = poll_role_string || "[parent]"
```

File: code/modules/mob/living/simple_animal/bot/bot.dm [hunk 3]
```diff
@@ -48,8 +46,8 @@
 	var/list/prev_access = list()
 
 	///Bot-related mode flags on the Bot indicating how they will act.
-	var/bot_mode_flags = BOT_MODE_ON | BOT_MODE_REMOTE_ENABLED | BOT_MODE_PAI_CONTROLLABLE
-//	Selections: BOT_MODE_ON | BOT_MODE_AUTOPATROL | BOT_MODE_REMOTE_ENABLED | BOT_MODE_PAI_CONTROLLABLE
+	var/bot_mode_flags = BOT_MODE_ON | BOT_MODE_REMOTE_ENABLED | BOT_MODE_GHOST_CONTROLLABLE
+//	Selections: BOT_MODE_ON | BOT_MODE_AUTOPATROL | BOT_MODE_REMOTE_ENABLED | BOT_MODE_GHOST_CONTROLLABLE
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/bot/bot.dm:50]: This should be above the bot_mode_flags variable, and also part of the `///` documentation

File: tgui/packages/tgui/interfaces/SimpleBot.tsx [hunk 4]
```diff
@@ -187,6 +162,23 @@ const SettingsDisplay = (props, context) => {
           />
         </Tooltip>
       </LabeledControls.Item>
+      {allow_possession && (
+        <LabeledControls.Item label="Personality">
+          <Tooltip
+            content={
+              !possession_enabled
+                ? 'Enables download of a unique personality.'
+                : 'Resets personality to factory default.'
```
Comment (Mothblocks) [inline tgui/packages/tgui/interfaces/SimpleBot.tsx:171]: Invert

**Review by Mothblocks — CHANGES_REQUESTED** [review]
To be clear, when does the ghost role thing popup? Is that just if someone requests it?

Also will this show up in the spawners menu? Because it should

**Comment by Mothblocks** [comment]
As for pAIs someone suggested letting them control bots in a nearby radius around their owner, which I think tracks with what I'd want pAIs to be. Don't have to do that in this PR though

**Comment by IndieanaJones** [comment]
To note my personal opinion on the emagging scenario by the way, I think they should be able to do whatever when emagged. We do have a traitor gimmicks where you make a bunch of ghost roles to help you wreck everything (xenobio gold slime and roboticist emagged borgs), and this would likely be much weaker but perhaps more entertaining.

**Comment by Jacquerel** [comment]
> Also will this show up in the spawners menu? Because it should

The spawners menu seems to very specifically only works for subtypes of `/obj/effect/mob_spawn/ghost_role` but I can see what I can do, I would also like for it to be there.

**Comment by GuillaumePrata** [comment]
Do they have any sort of lawsets? I couldn't find it with the search function at least.

Even if it was something as simple as "Perform X duties." or whatever is generic and broad, it would be cute to have a lawset, makes them feel more like proper silicons.

That also opens up a pandora box of silicon policy discussions so it might just not be worth the trouble...

**Comment by Jacquerel** [comment]
At the moment they just have spawn flavourtext which gives them a single goal to try to perform, which isn't really the same as a law. Usually "stuff which is printed to you on the screen" is considered to be binding in most cases but you're not wrong that there are likely to be some policy questions raised about this after it is merged (if it is merged).

**Comment by MrDoomBringer** [comment]
> Perhaps most controversially, any bots which are present on the station when the map loads will have this setting enabled by default.

this is awesome and great for the game imo, if anything is too easily abusable that's just exposing design flaws with the bots themselves imo

File: code/modules/mob/living/simple_animal/hostile/regalrat.dm [hunk 5]
```diff
@@ -35,50 +35,39 @@
...
 	var/datum/action/cooldown/riot/riot
+	///Should we request a mind immediately upon spawning?
+	var/poll_ghosts = FALSE
 
 /mob/living/simple_animal/hostile/regalrat/Initialize(mapload)
 	. = ..()
+	ADD_TRAIT(src, TRAIT_VENTCRAWLER_ALWAYS, INNATE_TRAIT)
+	AddElement(/datum/element/waddling)
+	AddComponent(\
+		/datum/component/ghost_direct_control,\
```
Comment (MrMelbert) [inline code/modules/mob/living/simple_animal/hostile/regalrat.dm:46]: Cargorilla should use this too, pretty much the same as the Rat

Comment (Jacquerel) [reply code/modules/mob/living/simple_animal/hostile/regalrat.dm:46]: Noticed that myself, already done in incoming commit 👍

File: code/_globalvars/lists/objects.dm [hunk 6]
```diff
@@ -60,7 +60,8 @@ GLOBAL_LIST_EMPTY(wire_name_directory)
 
 GLOBAL_LIST_EMPTY(ai_status_displays)
 
-GLOBAL_LIST_EMPTY(mob_spawners)     // All mob_spawn objects
+GLOBAL_LIST_EMPTY(mob_spawners) // All mob_spawn objects
+GLOBAL_LIST_EMPTY(joinable_mobs) // Mobs which you can click on to control them
```
Comment (MrMelbert) [inline code/_globalvars/lists/objects.dm:64]: Auto docs

File: code/datums/components/_component.dm [hunk 7]
```diff
@@ -52,11 +52,14 @@
...
-		return
-
+	switch(Initialize(arglist(arguments)))
+		if (COMPONENT_INCOMPATIBLE)
+			stack_trace("Incompatible [type] assigned to a [parent.type]! args: [json_encode(arguments)]")
+			qdel(src, TRUE, TRUE)
+			return
+		if(INITIALIZE_HINT_QDEL)
+			qdel(src, TRUE, TRUE)
+			return
```
Comment (MrMelbert) [inline code/datums/components/_component.dm:62]: Hmmm this smells like #76005 a little bit. Not sure we should be doing this.

Comment (Jacquerel) [reply code/datums/components/_component.dm:62]: I guess I can just repeatedly check for the global flag instead.

File: code/modules/mob/living/simple_animal/bot/bot.dm [hunk 8]
```diff
@@ -393,21 +429,10 @@
...
-					user.visible_message(span_notice("[user] uses [attacking_item] to pull [paicard] out of [initial(src.name)]!"),span_notice("You pull [paicard] out of [initial(src.name)] with [attacking_item]."))
-					ejectpai(user)
-	else
-		if(attacking_item.force) //if force is non-zero
-			do_sparks(5, TRUE, src)
-		..()
+		return
+	if(attacking_item.force) //if force is non-zero
+		do_sparks(5, TRUE, src)
+	return ..()
```
Comment (MrMelbert) [inline code/modules/mob/living/simple_animal/bot/bot.dm:435]: If you may please, move this to `attacked_by` (`.` being truthy means successful attack)

**Comment by MrMelbert** [comment]
Perhaps remove simplebots from the random human level intelligence event?

**Comment by Jacquerel** [comment]
> Perhaps remove simplebots from the random human level intelligence event?

I can do this but currently it selects simply from "all simple or basic mobs present on the map" if it fails to find something high priority, so that might just end up as "remove mules from high priority list"

**Comment by AdipemDragon** [comment]
Honestly, as a pAI main, I... _Genuinely_ don't know what to make of this? Like yeah, this'll be an actually interesting first step to modifying pAI's in a way that's not as aggro as other proposed changes; technically, for this PR, Nothing Changes in the grand scheme of things of the station running. _Sort of. Kind of not really?_ It's all just small things, really.

I just find it a bummer for the flavor, really, Like, "oh, who's this lil fella helpin me? Oh it's that downloaded pAI from earlier that I see all the time" versus picking up unnamed bots(or the named roundstart ones) as _some consciousness_, which could be anyone, who knows! It's effectively like swapping out pAI's for positronic brains, minus the material cost, so... 🤷‍♂️? (which puts the sentience granter for the mining bot in an interesting spot, but that's not here nor there)
Also, it'll sure be something to see those pAI cards that're roundstart in robotics on several maps no longer being used for their sentient bot use, that's for sure.

Like yeah, the goal'll eventually be to just make the p in pAI more pronounced, but man, it'll just be Not The Same in the end.
Oh well, we'll see what that's like when we get there.

**Comment by Exester509** [comment]
> Hygeinebots can't have pAIs inserted because their emagged version spews fire at whoever they try to clean.
> 
> Mulebots should probably not be included here, that sounds like a nightmare waiting to happen.
> 
> E: Turns out controlled mulebots can't run over people with a mind, so that's not a concern, but I do worry about trying to use the mulebot just to find out two ghost roles have taken them and fucked off to maint or something.

Pretty sure mulebots and honkbots need to be emagged in order to be nasty. Alternatively, the AI can disable their safety protocols for the same results. Haven't tested mulebots but it does work like that on honks

**Comment by Exester509** [comment]
> Do they have any sort of lawsets? I couldn't find it with the search function at least.
> 
> Even if it was something as simple as "Perform X duties." or whatever is generic and broad, it would be cute to have a lawset, makes them feel more like proper silicons.
> 
> That also opens up a pandora box of silicon policy discussions so it might just not be worth the trouble...

No, robots have no lawsets and never have. Even pAIs couldn't access theirs while in control. They're also extraordinarily limited and able to be killed from 90% of the PDAs on the station

**Comment by ATH1909** [comment]
an important distinction between these bots and pAI-controlled bots is that pAI-controlled bots spit you out as a card when you die in your bot shell, while non-pAI bots just die permanently if they die in their bot shell. this is a problem for knifebots, who can and will be randomly killed with no notice by a crewmember who doesn't like the fact they got stabbed. if you're a pAI, you can holoform out and ask a roboticist for a new body; as a pAI-less bot, you kind of have to hope that they decide to make another on their own.

this permadeath can also encourage spider or free golem-like behavior where someone who gets killed as a bot just immediately respawns as another and keeps going. you could do that before as a pAI by ghosting, but that's kind of a faux pax/really obvious shittery.

**Comment by ATH1909** [comment]
tldr; I don't like that this PR creates a discontinuity between your bot "lives", which basically resets the progress of your story whenever you die. The permadeath also makes escalation more of a headache.

**Comment by TheVekter** [comment]
> tldr; I don't like that this PR creates a discontinuity between your bot "lives", which basically resets the progress of your story whenever you die. The permadeath also makes escalation more of a headache.

You don't remember anything when you cone back as a pAI as it is, how is this any different?

**Comment by SomeoneYouProbablyKnow** [comment]
He means that you don't "come back" as a pai, you just get kicked out of the bot and try again. While here you actually "respawn". 

It's okay people will just fully destroy knife bots

**Comment by MrrFiish** [comment]
Okay but what happens if I want to get a pAI and put a knife on a cleanbot? Can't do it anymore at all?

**Comment by ATH1909** [comment]
> > tldr; I don't like that this PR creates a discontinuity between your bot "lives", which basically resets the progress of your story whenever you die. The permadeath also makes escalation more of a headache.
> 
> You don't remember anything when you cone back as a pAI as it is, how is this any different?

by "come back", I mean your pAI card (with you still alive in it) is ejected when the bot dies, so you can talk, engage holoform, walk away, etc. and get a new bot to possess from robotics

if someone picks you up and wipes you and you then redownload into a new card, yeah, you won't have memories of your previous life

**Comment by Jacquerel** [comment]
> Okay but what happens if I want to get a pAI and put a knife on a cleanbot? Can't do it anymore at all?

You can have a sapient cleanbot with a knife and/or a pAI, the systems just don't touch.

**Comment by Jacquerel** [comment]
> an important distinction between these bots and pAI-controlled bots is that pAI-controlled bots spit you out as a card when you die in your bot shell, while non-pAI bots just die permanently if they die in their bot shell. this is a problem for knifebots, who can and will be randomly killed with no notice by a crewmember who doesn't like the fact they got stabbed.

I'll be honest this doesn't sound like a big problem to me.
If you're playing a respawn role, harm someone, and they kill you then that is simply things working as intended. Your being there on the sufferance of the crew is a feature not a bug, I think?

**Comment by ATH1909** [comment]
> I'll be honest this doesn't sound like a big problem to me. If you're playing a respawn role, harm someone, and they kill you then that is simply things working as intended. Your being there on the sufferance of the crew is a feature not a bug, I think?

knifebots automatically stab anyone who steps over them. this is how they collect titles, but it also sometimes pisses people off to an irrational degree. this is fine (and funny), but largely because the consequences are just you losing a knifebot body (and robotics maybe being forbidden from making you a new one) and not you getting round removed.

**Comment by Mothblocks** [comment]
get them to be in the spawners menu later so that people actually discover this

**Comment by chemistrymain2** [comment]
> > I just don't really understand the point of caring about the name that much? I feel like the actual gameplay itself should be what you're focusing on. I don't really know how to explain this any better, but it's like you're basically just creating a new tool called dwAI (Do-whatever AI) that can be used instead of pAI. In the end, all this change really accomplishes is removing the feature of being a murder bot like I said before. It may solve the problem of the role not being named correctly, but it doesn't solve the actual problem of people taking over bots and fucking off as them.
> 
> We don't care about people taking over bots and fucking off as them, we care about people taking pAI cards and fucking off as them. The problem you're identifying isn't the problem I am setting out to solve, nor frankly do I think it is a big deal. If people don't want their cleanbot to wander off they'll just not turn on this feature.

You're missing the point that it ruins antag bots.

**Comment by Exester509** [comment]
> > > I just don't really understand the point of caring about the name that much? I feel like the actual gameplay itself should be what you're focusing on. I don't really know how to explain this any better, but it's like you're basically just creating a new tool called dwAI (Do-whatever AI) that can be used instead of pAI. In the end, all this change really accomplishes is removing the feature of being a murder bot like I said before. It may solve the problem of the role not being named correctly, but it doesn't solve the actual problem of people taking over bots and fucking off as them.
> > 
> > 
> > We don't care about people taking over bots and fucking off as them, we care about people taking pAI cards and fucking off as them. The problem you're identifying isn't the problem I am setting out to solve, nor frankly do I think it is a big deal. If people don't want their cleanbot to wander off they'll just not turn on this feature.
> 
> You're missing the point that it ruins antag bots.

-"It doesn't solve the actual problem of people taking over bots and fucking off as them"
As you were told, turn sentience off if that bothers you so much. Anyone can do it from RoboTact
-"It ruins antag bots"
pAIs inserted into a robot don't really have any laws, nor can you access them in any way. Robots have the exact same freedom as before, if AI chooses to turn safeties off for a cleanbot you can still lube hallways if you want, or run around downing people as a honkbot. You rely on others to make you able to do any of those things and aren't really bound by anything

**Comment by BlueMemesauce** [comment]
They're bound by flavortext which cannot be changed, and according to server rules flavortext must be followed for ghost spawns. If they get emagged, they literally can't use their dangerous abilities do anything since it would be a violation of their flavortext.

**Comment by optimumtact** [comment]
@Mothblocks this is a very bad change, these are effectively now proto drones and suffer the exact same issues of admin enforcement and policy.

There is no way that this goes any differently to drones.

**Comment by TheVekter** [comment]
> @Mothblocks this is a very bad change, these are effectively now proto drones and suffer the exact same issues of admin enforcement and policy.
> 
> There is no way that this goes any differently to drones.

I don't really agree because pretty much the only actual antagonistic things non-tampered bots can do is slip people, drag things, and run them over as mulebots (but that's being restricted to an emag/hacking only thing so the fault is easier to track).

I guess there's knife cleanbots but they do very little damage.

**Comment by optimumtact** [comment]
Your comment admits to it being a problem in the same breath as denying it, which is frankly a pretty incredible effort.

**Comment by optimumtact** [comment]
It will not be the last problem you have, drones were fine in the beginning to, but the sandbox is endlessly complex and there will always be unintended interactions and the playerbase will find them and abuse them, over and over and over, until whatever admins are on the team in 2 years will beg us to remove bots being playable or introduce such massive restrictions as to achieve the same thing, because they're already busy enough preventing the actual player mobs and antags from griefing as well.

**Comment by Archie700** [comment]
Drones had the massive issue that they were basically very fast, very small mobs which could hold things and shoot guns, which necessitated a lot of restrictions that eventually left them gutted. 

Bots have heavily restricted functions compared to drones, so they should be comparatively easier to deal with.

**Comment by optimumtact** [comment]
he lied confidently

**Comment by TheSilverNuke** [comment]
I like the fact you don't need pAI's but the fact you can't play as an emagged bot is a serious bummer.
